packages/common/src/audius-query/hooks/usePaginatedQuery.ts

```diff
@@ -59,55 +59,61 @@ export const useAllPaginatedQuery = <
   const { pageSize, ...queryHookOptions } = options
   const [page, setPage] = useState(0)
   const [allData, setAllData] = useState<Data[]>([])
-  const args = {
-    ...baseArgs,
-    limit: pageSize,
-    offset: page * pageSize
-  } as ArgsType
-  const result = useQueryHook(args, queryHookOptions)
 
   useCustomCompareEffect(
     () => {
       setAllData([])
+      setPage(0)
       setLoadingMore(false)
     },
     [baseArgs],
     isEqual
   )
 
+  const args = {
+    ...baseArgs,
+    limit: pageSize,
+    offset: page * pageSize
+  } as ArgsType
+  const result = useQueryHook(args, queryHookOptions)
+
   useCustomCompareEffect(
     () => {
+      if (result.status !== Status.SUCCESS) return
+      setAllData((allData) => [...allData, ...result.data])
       if (!statusIsNotFinalized(result.status)) {
         setLoadingMore(false)
       }
-      if (result.status !== Status.SUCCESS) return
-      setAllData((allData) => [...allData, ...result.data])
     },
     [result.status, args],
     isEqual
   )
 
   const notError = result.status !== Status.ERROR
+  const notStillLoadingCurrentPage =
+    !loadingMore && !(result.status === Status.LOADING)
   const notStarted = result.status === Status.IDLE && allData.length === 0
   const hasNotFetched = !result.data && result.status !== Status.SUCCESS
   const fetchedFullPreviousPage = result.data?.length === pageSize
 
   const hasMore =
-    notError && (notStarted || hasNotFetched || fetchedFullPreviousPage)
+    notError &&
+    notStillLoadingCurrentPage &&
```

Comment:
Need this check here to prevent calling loadMore in the wrong order of operations on base params change